File: task3/java/SubTask3.java
import java.util.Scanner;

public class SubTask3 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the number of rows: ");
        int n = scanner.nextInt();

        System.out.println("Pattern 1");

        // Upper half of the diamond
        for (int i = 1; i <= (n + 1) / 2; i++) {
            System.out.print(" ".repeat((n + 1) / 2 - i));
            System.out.println("*".repeat((i * 2) - 1));
        }

        // Lower half of the diamond
        for (int i = (n + 1) / 2 + 1; i <= n; i++) {
            System.out.print(" ".repeat(i - (n + 1) / 2));
            System.out.println("*".repeat((n + 1 - i) * 2 - 1));
        }

        scanner.close();
    }
}

File: task3/go/SubTask3.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	var n int
	fmt.Print("Enter the number of rows: ")
	fmt.Scan(&n)

	fmt.Println("Pattern 1")

	// Upper half of the diamond
	for i := 1; i <= (n+1)/2; i() {
		fmt.Print(strings.Repeat(" ", (n+1)/2-i))
		fmt.Println(strings.Repeat("*", (i*2)-1))
	}

	// Lower half of the diamond
	for i := (n + 1) / 2 + 1; i <= n; i() {
		fmt.Print(strings.Repeat(" ", i-(n+1)/2))
		fmt.Println(strings.Repeat("*", (n+1-i)*2-1))
	}
}
